### PR TITLE
br: fix compatibility issue with concurrent ddl

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/36453

Problem Summary:
After concurrent DDL feature implemented. v6.2 cluster will use a new way(ddl table) to execute ddls. but when using v6.2 br to restore cluster to v6.1 doesn't have such ddl tables.

### What is changed and how it works?
This PR add a check. when using br (> v6.2) restore cluster (< v6.1). br will disable the switch of concurrent ddl.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
截屏2022-07-22 下午4 07 08

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/version/version.go [hunk 1]
```diff
@@ -151,6 +152,17 @@ func CheckVersionForBRPiTR(s *metapb.Store, tikvVersion *semver.Version) error {
 	return nil
 }
 
+// CheckVersionForDDL checks whether we use queue or table to execute ddl during restore.
+func CheckVersionForDDL(s *metapb.Store, tikvVersion *semver.Version) error {
+	// use tikvVersion instead of tidbVersion since br doesn't have mysql client to connect tidb.
+	if tikvVersion.Major < 6 || (tikvVersion.Major == 6 && tikvVersion.Minor <= 1) {
```
Comment (xiongjiwei) [inline br/pkg/version/version.go:158]: can use compare method

File: br/pkg/version/version.go [hunk 2]
```diff
@@ -151,6 +152,17 @@ func CheckVersionForBRPiTR(s *metapb.Store, tikvVersion *semver.Version) error {
...
+		log.Info("detected the old version of tidb cluster. set enable concurrent ddl to false")
+		variable.EnableConcurrentDDL.Store(false)
```
Comment (xiongjiwei) [inline br/pkg/version/version.go:160]: no need to set it back?

Comment (3pointer) [reply br/pkg/version/version.go:160]: yes, this variable only works when br exists.

**Review by D3Hunter — APPROVED** [review]
lgtm

File: br/pkg/version/version.go [hunk 3]
```diff
@@ -151,6 +152,18 @@ func CheckVersionForBRPiTR(s *metapb.Store, tikvVersion *semver.Version) error {
 	return nil
 }
 
+// CheckVersionForDDL checks whether we use queue or table to execute ddl during restore.
+func CheckVersionForDDL(s *metapb.Store, tikvVersion *semver.Version) error {
+	// use tikvVersion instead of tidbVersion since br doesn't have mysql client to connect tidb.
+	requireVersion := semver.New("6.2.0")
```
Comment (xiongjiwei) [inline br/pkg/version/version.go:158]:
```suggestion
	requireVersion := semver.New("6.2.0-alpha")
```

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**